storage/src/main/java/tech/pegasys/teku/storage/store/StoreTransactionUpdates.java

```diff
@@ -118,16 +118,24 @@ public void applyToStore(final Store store, final UpdateResult updateResult) {
 
     // Update finalized data
     finalizedChainData.ifPresent(
-        finalizedData -> store.finalizedAnchor = finalizedData.getLatestFinalized());
+        finalizedData -> {
+          store.maybeEpochStates.ifPresent(
```

Comment:
can we add some comments to cover the reasoning? or move this in a method with a meaningful name

Reply:
i pretty much rewrote all of this should be clearer now